webservices/resources/legal.py

```diff
@@ -22,6 +24,45 @@ def get(self, ao_no, **kwargs):
         return results
 
 
+phrase_regex = re.compile('"(?P<phrase>[^"]*)"')
+def parse_query_string(query):
+    """Parse phrases from a query string for exact matches e.g. "independent agency"."""
+
+    def _parse_query_string(query):
+        """Recursively pull out terms and phrases from query. Each pass pulls
+        out terms leading up to the phrase as well as the phrase itself. Then
+        it processes the remaining string."""
+
+        if not query:
+            return ([], [])
+
+        match = phrase_regex.search(query)
```

Comment:
You've probably thought through this more than I have, but is recursion really necessary here? Why not just use `re.findall(regex, str)`?